### PR TITLE
ditch multiple getEntitySetId and getPropertyTypeId calls

### PR DESCRIPTION
**Pull request by anzioka**
PR: get rid of this -> 
Screen Shot 2020-05-22 at 3 21 09 PM

### CODE REVIEW

File: src/main/java/com/openlattice/chronicle/constants/EdmConstants.java [hunk 1]
```diff
@@ -29,14 +33,22 @@ private EdmConstants() {
...
+            USED_BY_ENTITY_SET_NAME,
+            PARTICIPATED_IN_AESN,
+            QUESTIONNAIRE_ENTITY_SET_NAME,
+            QUESTIONS_ENTITY_SET_NAME,
+            ANSWERS_ENTITY_SET_NAME,
+            RESPONDS_WITH_ENTITY_SET_NAME,
+            ADDRESSES_ENTITY_SET_NAME,
+            NOTIFICATION_ENTITY_SET_NAME,
+            PART_OF_ENTITY_SET_NAME
+    ) );
```
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/constants/EdmConstants.java:51]: Use `ImmutableSet`:

```
public static final ImmutableSet<String> ENTITY_SET_NAMES = ImmutableSet.of(
  ADDRESSES_ENTITY_SET_NAME,
  ...
);
```

Also, please sort.

File: src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java [hunk 2]
```diff
@@ -151,38 +119,20 @@ public ApiClient load( Class<?> key ) throws Exception {
...
-        versionPTID = edmApi.getPropertyTypeId( VERSION_FQN.getNamespace(), VERSION_FQN.getName() );
-        modelPTID = edmApi.getPropertyTypeId( MODEL_FQN.getNamespace(), MODEL_FQN.getName() );
-        dateTimePTID = edmApi.getPropertyTypeId( DATE_TIME_FQN.getNamespace(), DATE_TIME_FQN.getName() );
-        titlePTID = edmApi.getPropertyTypeId( TITLE_FQN.getNamespace(), TITLE_FQN.getName() );
-        fullNamePTID = edmApi.getPropertyTypeId( FULL_NAME_FQN.getNamespace(), FULL_NAME_FQN.getName() );
-        recordTypePTID = edmApi.getPropertyTypeId( RECORD_TYPE_FQN.getNamespace(), RECORD_TYPE_FQN.getName() );
-        startDateTimePTID = edmApi.getPropertyTypeId( START_DATE_TIME.getNamespace(), START_DATE_TIME.getName() );
-        dataKey = edmApi.getEntityType( entitySetsApi.getEntitySet( dataESID ).getEntityTypeId() ).getKey();
+        // get entity setId map
+        entitySetIdMap = entitySetsApi.getEntitySetIds( ENTITY_SET_NAMES );
```
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java:123]: Use `ImmutableMap`:

```
private final ImmutableMap<String, UUID> entitySetIdMap;
entitySetIdMap = ImmutableMap.copyOf( entitySetsApi.getEntitySetIds( ENTITY_SET_NAMES ) );
```

File: src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java [hunk 3]
```diff
@@ -151,38 +119,20 @@ public ApiClient load( Class<?> key ) throws Exception {
...
+
+        // get propertyTypeId map
+        Iterable<PropertyType> propertyTypes = edmApi.getPropertyTypes();
+        propertyTypeIdMap = StreamSupport
+                .stream( propertyTypes.spliterator(), false )
+                .collect( Collectors.toMap(
+                        propertyType -> propertyType.getType(),
+                        propertyType -> propertyType.getId()
+                ) );
```
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java:132]: Use `ImmutableMap`. Also, we should also have another map to remove duplicate logic from `getParticipantDataHelper()`:

```
private final ImmutableMap<UUID, PropertyType> propertyTypesById;
private final ImmutableMap<FullQualifiedName, UUID> propertyTypeIdsByFQN;

propertyTypesById = StreamSupport
    .stream( edmApi.getPropertyTypes().spliterator(), false )
    .collect( ImmutableMap.toImmutableMap( PropertyType::getId, Function.identity() ) );

propertyTypeIdsByFQN = propertyTypesById
    .values()
    .stream()
    .collect( ImmutableMap.toImmutableMap( PropertyType::getType, PropertyType::getId ) );
```

File: src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java [hunk 4]
```diff
@@ -90,48 +90,17 @@
...
-    private final UUID titlePTID;
-    private final UUID fullNamePTID;
-    private final UUID recordTypePTID;
-    private final UUID startDateTimePTID;
-    private final UUID durationPTID;
+    private final ImmutableMap<UUID, PropertyType> propertyTypesById;
+    private final ImmutableMap<String, UUID>                entitySetIdMap;
+    private final Map<FullQualifiedName, UUID>     propertyTypeIdsByFQN;
+    private final String                           username;
+    private final String                           password;
```
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java:97]: Fix alignment.

File: src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java [hunk 5]
```diff
@@ -90,48 +90,17 @@
...
-    private final UUID modelPTID;
-    private final UUID userAppsESID;
-    private final UUID titlePTID;
-    private final UUID fullNamePTID;
-    private final UUID recordTypePTID;
-    private final UUID startDateTimePTID;
-    private final UUID durationPTID;
+    private final ImmutableMap<UUID, PropertyType> propertyTypesById;
+    private final ImmutableMap<String, UUID>                entitySetIdMap;
+    private final Map<FullQualifiedName, UUID>     propertyTypeIdsByFQN;
```
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java:95]: This is also `ImmutableMap`.

Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java:95]: oh yes!

**Comment by jokedurnez** [comment]
Just a general question about this: is this forward thinking for when we have AppConfigurables?  The property types should be loaded at start, but the entitysets will have to be queried as data/queries come in right?

**Comment by anzioka** [comment]
> Just a general question about this: is this forward thinking for when we have AppConfigurables? The property types should be loaded at start, but the entitysets will have to be queried as data/queries come in right?

No, this is for removing clutter / replacing multiple getEntitySetId and getPropertyTypeId calls with a single call. That's a good point though.

**Comment by jokedurnez** [comment]
Ok sg. Maybe this will even make it easier to later refactor.

**Pull request merged**